### PR TITLE
[linear-gradient] add doc comments in source

### PR DESCRIPTION
**Pull request by Simek**
# Why

To utilize the auto generation feature for API section the package source code must contain the relevant documentation.

Refs #12239

Fixes ENG-825

# How

This PR moves the current documentation of `LinearGradient` API - https://docs.expo.io/versions/latest/sdk/linear-gradient/ - (with minor tweaks and addition) to the package source code. Also the `[number, number]` raw type used in component helper method has been replace with `NativeLinearGradientPoint`.

The changeset also includes the following improvements to the autogenerated content rendering code in the docs:
* improved prop types extraction - `LinearGradientPoint` was not rendered at all,
* improved prop styling and additional optional flag for the props (seen on the preview screenshot).

---

So far (in the two component which have been converted) the `Props` were not stylised in the way which will adds them to the right sidebar, which prevents scrolling on page enter, from the right side, to the selected/searched prop. What do you think about changing it?

# Test Plan

The generated content and changes to the APISection components have been tested running docs website on `localhost`.

# Preview

Screenshot 2021-04-20 at 17 43 42

# Checklist

- [X] Documentation is up to date to reflect these changes (eg: https://docs.expo.io and README.md).
- [X] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [X] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

CC: @bbarthec @brentvatne @jonsamp

### CODE REVIEW

**Review by brentvatne — APPROVED** [review]
looks nice 😎

File: docs/components/plugins/api/APISectionProps.tsx [hunk 1]
```diff
@@ -44,7 +48,7 @@ const renderInheritedProp = (ip: TypeDeclarationData) => {
 };
 
 const renderInheritedProps = (data: TypeDeclarationData[]): JSX.Element | undefined => {
-  const inheritedProps = data.filter((ip: TypeDeclarationData) => ip.type === 'reference');
+  const inheritedProps = data?.filter((ip: TypeDeclarationData) => ip.type === 'reference') || [];
```
Comment (bbarthec) [inline docs/components/plugins/api/APISectionProps.tsx:51]:
```suggestion
  const inheritedProps = data?.filter((ip: TypeDeclarationData) => ip.type === 'reference') ?? [];
```

**Review by bbarthec — APPROVED** [review]
💪

**Review by tsapeta — APPROVED** [review]

**Comment by tsapeta** [comment]
Some CI checks are failing, but looks like they are unrelated to this PR, so I'm gonna merge it 😉

**Comment by Simek** [comment]
> Some CI checks are failing, but looks like they are unrelated to this PR, so I'm gonna merge it 😉

`docs` failure might be related to the update to Webpack 5 in Next.

**Pull request merged**